nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -150,548 +145,320 @@
-        splits.clear();
-        for (int i = 1; i <= newList.size(); i++) {
-            FlowFile ff = newList.get(i - 1);
-            final Map<String, String> attributes = new HashMap<>();
-            attributes.put(FRAGMENT_ID, fragmentId);
-            attributes.put(FRAGMENT_INDEX, String.valueOf(i));
-            attributes.put(FRAGMENT_COUNT, String.valueOf(newList.size()));
-            attributes.put(SEGMENT_ORIGINAL_FILENAME, originalFilename);
-            FlowFile newFF = session.putAllAttributes(ff, attributes);
-            splits.add(newFF);
-        }
+    /**
+     *
+     */
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        List<ValidationResult> results = new ArrayList<>();
+        boolean invalidState = (validationContext.getProperty(LINE_SPLIT_COUNT).asInteger() == 0
+                && !validationContext.getProperty(FRAGMENT_MAX_SIZE).isSet());
+        results.add(new ValidationResult.Builder().subject("Maximum Fragment Size").valid(!invalidState)
+                .explanation("Property must be specified when Line Split Count is 0").build());
+        return results;
     }
 
-    private static class SplitInfo {
-
-        public long offsetBytes;
-        public long lengthBytes;
-        public long lengthLines;
-        public long bufferedBytes;
-        public boolean endOfStream;
-
-        public SplitInfo() {
-            this.offsetBytes = 0L;
-            this.lengthBytes = 0L;
-            this.lengthLines = 0L;
-            this.bufferedBytes = 0L;
-            this.endOfStream = false;
-        }
+    /**
+     *
+     */
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return Collections.unmodifiableList(properties);
     }
 
-    public static class EndOfLineBuffer {
-        private static final byte CARRIAGE_RETURN = (byte) '\r';
-        private static final byte NEWLINE = (byte) '\n';
-
-        private final BitSet buffer = new BitSet();
-        private int index = 0;
-
-        public void clear() {
-            index = 0;
-        }
-
-        public void addEndOfLine(final boolean carriageReturn, final boolean newLine) {
-            buffer.set(index++, carriageReturn);
-            buffer.set(index++, newLine);
-        }
-
-        private void drainTo(final OutputStream out) throws IOException {
-            for (int i = 0; i < index; i += 2) {
-                final boolean cr = buffer.get(i);
-                final boolean nl = buffer.get(i + 1);
-
-                // we've consumed all data in the buffer
-                if (!cr && !nl) {
-                    return;
-                }
-
-                if (cr) {
-                    out.write(CARRIAGE_RETURN);
+    /**
+     *
+     */
+   private FlowFile updateAttributes(ProcessSession session, FlowFile splitFlowFile, long lCount, long fSize, String fId, int fIdx, int fCount, String origFname) {
```

Comment:
should avoid abbreviations of variable names - nifi tends to spell everything out